### DISCUSSION TITLE
Reading program from Hub

### DISCUSSION THREAD

**Discussion by rbeckett75** [pybricks/support#1308] [General]
I was excited to try out the new blocks coding.. and the side panel/documentation wasn't loading (was getting a connection error)
So I cleared browser cache - which fixed it...

...and also wiped out all my coding documents!

I'm not fussed too much about most of them - I did do a backup a month ago

However the main one I had been working on - wasn't backed up (my fault!)

It's still on the hub that I'm using - is there a way to download it off it?

Thanks!

**Comment by dlech** [comment]
Unfortunately, no there isn't. There is no Bluetooth command to do this and even if there was, the version of the program on the hub is a binary `.mpy` file and not the original text `.py` file.

**Reply by rbeckett75** [reply]
I was suspecting that might be the case! 

in any case, I found a newer backup with my missing code this morning! :)

Next question/comments on this whole topic:

1. Do the hubs have enough storage, that you could enable an option to simply upload the .py text file too? most programs, zipped would be under a few kb?
2. Any future plans to integrate the IDE with cloud storage?  would be great if I could link it to my Google Drive/OneDrive, and let it save there instead


The Block coding does look great! had a play last night, and I think I managed to recreate my (fairly simple) program last night.. I'll be testing it out after work today!
Really looking forward to trying out the Hub to Hub communication, as I struggled to get my head around it with regular code... 
I think the best thing about the block style, is how it still shows the code - it will be an excellent way to learn!

Thanks for all your great work!

**Reply by dlech** [reply]
> 1. Do the hubs have enough storage, that you could enable an option to simply upload the .py text file too? most programs, zipped would be under a few kb?

On hubs with external flash (SPIKE/MINDSTORMS), we could do this if we were using the external flash, but currently we don't touch the external flash.

> 2. Any future plans to integrate the IDE with cloud storage? would be great if I could link it to my Google Drive/OneDrive, and let it save there instead

No concrete plans currently.  Related discussion: https://github.com/pybricks/support/issues/1004

**Reply by laurensvalk** [reply]
> in any case, I found a newer backup with my missing code this morning! :)

I'm so glad to hear it! :tada: 

> The Block coding does look great! had a play last night, and I think I managed to recreate my (fairly simple) program last night.. I'll be testing it out after work today! Really looking forward to trying out the Hub to Hub communication, as I struggled to get my head around it with regular code... I think the best thing about the block style, is how it still shows the code - it will be an excellent way to learn!
> 
> Thanks for all your great work!

Thank you. I'm glad people are finding their way to the blocks and trying it out. Looking forward to seeing what you'll make with the communicating hubs.

> Any future plans to integrate the IDE with cloud storage? would be great if I could link it to my Google Drive/OneDrive, and let it save there instead

This is a great idea, and it would be especially neat for teams using it. I wonder how live synchronization could work to avoid conflicts between team members working at the same time.

**Comment by laurensvalk** [comment]
Thanks @rbeckett75 for signing up! Sorry about the issues with the docs. I'll push out a fix as soon as possible.

Hopefully with blocks you'll be able to rebuild your program quickly. Please do let me know if we can help you with that :smile:
